webpack/components/extensions/HostDetails/Tabs/PackagesTab.js

```diff
@@ -73,20 +89,117 @@ export const PackagesTab = () => {
   const response = useSelector(state => selectAPIResponse(state, HOST_PACKAGES_KEY));
   const { results, ...metadata } = response;
   const status = useSelector(state => selectHostPackagesStatus(state));
+  const {
+    selectOne, isSelected, searchQuery, selectedCount, isSelectable,
+    updateSearchQuery, selectNone, fetchBulkParams, ...selectAll
+  } = useBulkSelect({
+    results,
+    metadata,
+    idColumn: 'id',
+    isSelectable: result => result.id,
+  });
 
   if (!hostId) return <Skeleton />;
+
   const handleInstallPackagesClick = () => {
     setIsBulkActionOpen(false);
     setIsModalOpen(true);
   };
 
-  const rowActions = [
-    {
-      title: __('Upgrade via remote execution'), disabled: true,
-    },
-    {
-      title: __('Upgrade via customized remote execution'), disabled: true,
-    },
+  const removePackageViaRemoteExecution = packageName => dispatch(removePackage({
+    hostname,
+    packageName,
+  }));
+
+  const removePackageViaKatelloAgent = (packageName) => {
+    selectNone();
+    dispatch(removeViaKatelloAgent(hostId, { packages: [packageName] }));
+  };
+
+  const removePackagesViaRemoteExecution = () => {
+    const selected = fetchBulkParams();
+    setIsBulkActionOpen(false);
+    selectNone();
+    dispatch(removePackages({ hostname, search: selected }));
+  };
+
+  const katelloAgentAvailable = (contentFacet.katelloAgentInstalled &&
+    contentFacet.katelloAgentEnabled);
+
+  const removeBulk = () => {
+    if (contentFacet.remoteExecutionByDefault || !katelloAgentAvailable) {
+      removePackagesViaRemoteExecution();
+    }
+  };
+
+  const removeKatelloPackage = (packageName) => {
+    if (contentFacet.remoteExecutionByDefault || !katelloAgentAvailable) {
```

Comment:
This was just merged; I suggest reusing it here: https://github.com/Katello/katello/blob/d433a901c0247b699ff742791367ecd73dd304e5/webpack/components/extensions/HostDetails/hostDetailsHelpers.js#L6